lib/common/db-renderable-content.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = dbRenderableContentServiceFactory;
+
+dbRenderableContentServiceFactory.$provide = 'DbRenderableContent';
+dbRenderableContentServiceFactory.$inject = [
+    'FileLoader',
+    'Constants',
+    'Promise',
+    '_',
+    'Services.Waterline',
+    'Logger',
+    'ejs'
+];
+
+/**
+ * dbRenderableContentServiceFactory provides the template-service singleton object.
+ * @private
+ * @param  {FileLoader} FileLoader A FileLoader class for extension.
+ * @param  {configuration} configuration      An instance of the configuration configuration object.
+ * @return {TemplateService}            An instance of the TemplateService.
+ */
+function dbRenderableContentServiceFactory(
+    FileLoader, 
+    Constants, 
+    Promise, 
+    _, 
+    waterline, 
+    Logger, 
+    ejs ) 
+{
+    var logger = Logger.initialize(dbRenderableContentServiceFactory);
+
+    function DbRenderableContentService () {
+        this.loader = new FileLoader();
+    }
+
+    DbRenderableContentService.prototype.load = function load() {
+        var self = this;
+
+        return this.loader.getAll(
+            self.directory
+        ).then(function (templates) {
+            var promises = _.map(templates, function (contents, name) {
+                return self.put(name, contents.toString());
+            });
+
+            return Promise.all(promises);
+        }).catch(function (err) {
+            if (process.env.NODE_ENV !== 'test') {
+                logger.error(
+                    'Unable to load ' + self.collectionName + ' from ' + self.directory + '.',
+                    {
+                        error: err
+                    }
+                );
+            }
+        });
+    };
+
+    DbRenderableContentService.prototype.put = function (file, contents, scope) {
+        var self = this;
+        scope = scope || Constants.Scope.Global;
+        return Promise.resolve(
+            waterline[self.collectionName].findOne({ name: file, scope: scope })
+            .then(function(doc) {
+                if (!_.isEmpty(doc)) {
+                    return waterline[self.collectionName].update(
+                        { name: file, scope: scope },
+                        { contents :contents }
+                    );
+                } else {
+                    return waterline[self.collectionName].create(
+                        {
+                            name: file,
+                            contents: contents,
+                            scope: scope
+                        }
+                    );
+                }
+            })
+        );
+    };
+
+    DbRenderableContentService.prototype.get = function (name, scope) {
```

Comment:
I'd expect add some JSDoc for this function, this function contains some priority ranking, need some special description for it.